### PR TITLE
Introduce WebkitHeadless browser testing

### PR DESCRIPTION
**Pull request by dlarocque**
Introduce WebKitHeadless browser testing, using the [`karma-webkit-launcher`](https://github.com/google/karma-webkit-launcher) plugin. The plugin uses [playwright](https://github.com/microsoft/playwright) to launch it's `WebkitHeadless` browser, and uses that browser to run karma tests on.

This is significantly better than the proposed solution in #8396, where we forked the old `karma-safari-launcher` plugin, because it uses a headless browser. This means that
- We don't need to manually enable Full Disk Access on Safari to the host process
- The headless browser process will exit after tests are complete
- There are no browser popups when running the tests, since there's no head to pop up

**How to run locally:**
```bash
yarn install # make sure the new packages are installed
npx playwright install webkit # install playwright's webkit browser
BROWSERS='WebkitHeadless' yarn test:browser # or any other command that launches a browser for karma tests
```

**Testing:**

I ran all of the browser tests against WebkitHeadless locally, and did not encounter any unexpected failures (the flakes in firestore still occur). Everything works as expected.

When running the same tests in CI, I noticed that several packages are now flaky (seemingly only in CI?): installations, storage-compat, database, database-compat, app-check. It's not clear why these are suddenly only failing in the headless webkit browser in CI, so let's skip them for now, and add a `TODO` to fix this later.

### CODE REVIEW

File: .github/workflows/test-changed.yml [hunk 1]
```diff
@@ -78,3 +78,41 @@ jobs:
...
+      - name: Test setup and yarn install
+        run: |
+          cp config/ci.config.json config/project.json
+          yarn
+          npx playwright install webkit
+      - name: build
+        run: yarn build:changed core
+      - name: Run tests on changed packages
+        run: |
+          npx lerna run test:browser --scope @firebase/database-compat
```
Comment (hsubox76) [inline .github/workflows/test-changed.yml:105]: ?

Comment (dlarocque) [reply .github/workflows/test-changed.yml]: I made this change to help me identify the cause of the flakes. I've reverted all these changes.

File: .github/workflows/test-changed-firestore.yml [hunk 2]
```diff
@@ -16,7 +16,7 @@ name: Test Firestore
 
 on:
   workflow_dispatch:
-  pull_request:
```
Comment (hsubox76) [inline .github/workflows/test-changed-firestore.yml:19]: Why this change?

File: .github/workflows/test-all.yml [hunk 3]
```diff
@@ -91,6 +91,11 @@ jobs:
       run: echo "FIREBASE_CI_TEST_START_TIME=$(date +%s)" >> $GITHUB_ENV
     - name: Run unit tests
       # Ignore auth and firestore since they're handled in their own separate jobs.
+      # TODO (dlarocque): Fix flakes in installations, database, database-compat, storage-compat,
```
Comment (hsubox76) [inline .github/workflows/test-all.yml:94]: Isn't this workflow file all Chrome? Also isn't the skipping logic in testConfig.ts?

File: .github/workflows/test-changed-firestore.yml (hunk 2 above)
Comment (dlarocque) [reply .github/workflows/test-changed-firestore.yml:19]: I made this change so that I can test the workflow on my branch. I'll revert it

File: .github/workflows/test-all.yml (hunk 3 above)
Comment (dlarocque) [reply .github/workflows/test-all.yml:94]: Yeah this comment shouldn't be here- and there's already a comment in the test config so I'll just remove this.

**Review by DellaBitta — APPROVED** [review]

**Review by hsubox76 — APPROVED** [review]
